src/Quizzer.API.DAL/UnitOfWork/IUnitOfWork.cs

```diff
@@ -11,10 +13,25 @@ public interface IUnitOfWork : IDisposable {
         /// <returns></returns>
         IRepository<TEntity> GetRepository<TEntity>()
             where TEntity : class, IEntity;
-           
+
         /// <summary>
         /// Makes changes on entities in database
         /// </summary>
         void Commit();
+
+        /// <summary>
+        /// Discards changes
+        /// </summary>
+        public void Dispose();
+
+        /// <summary>
+        /// Makes changes on entities in database
+        /// </summary>
+        public Task CommitAsync();
+
+        /// <summary>
+        /// Discards changes
+        /// </summary>
+        public ValueTask DisposeAsync();
```

Comment:
I do not see you are using `IAsyncDisposable` interface in uow, the runtime will probably not trigger this method...

Reply:
`IUnitOfWork` should inherit `IAsyncDisposable` interface which contains `DisposeAsync`, during runtime, GC also checks all instances which inherits dispose async and when the object lifetime is at it's end, it will call it implicitly...

Reply:
removed.